### ISSUE TITLE
Storage Blob Service ContainerOperationFailure

### ISSUE THREAD

**Issue by yupwei68** [Azure/azure-rest-api-specs#11237]
1. PUT /subscriptions/.../resourceGroups/.../providers/Microsoft.Storage/storageAccounts/.../blobServices/default?api-version=2019-06-01

{"properties":{"cors":{"corsRules":[{"allowedOrigins":["http://www.example.com"],"allowedMethods":["GET","PUT","PATCH"],"maxAgeInSeconds":500,"exposedHeaders":["x-tempo-*"],"allowedHeaders":["x-tempo-*"]}]},"deleteRetentionPolicy":{"enabled":true,"days":300},"isVersioningEnabled":true,"changeFeed":{"enabled":true},"restorePolicy":{"days":5,"enabled":true}}}

200 OK

2. PUT /subscriptions/.../resourceGroups/.../providers/Microsoft.Storage/storageAccounts/.../blobServices/default?api-version=2019-06-01

{"properties":{"cors":{"corsRules":[{"allowedOrigins":["http://www.example.com"],"allowedMethods":["GET"],"maxAgeInSeconds":2000000000,"exposedHeaders":["x-tempo-*","x-method-*"],"allowedHeaders":["*"]},{"allowedOrigins":["http://www.test.com"],"allowedMethods":["PUT"],"maxAgeInSeconds":1000,"exposedHeaders":["x-tempo-*"],"allowedHeaders":["*"]}]},"deleteRetentionPolicy":{"enabled":true,"days":7},"isVersioningEnabled":false,"changeFeed":{"enabled":true},"restorePolicy":{"enabled":false}}}

413 Preconditional Failed 
{
    "error": {
        "code": "ContainerOperationFailure",
        "message": "The condition specified using HTTP conditional header(s) is not met.\nRequestId:13ccb51b-301e-006c-0f92-a3728a000000\nTime:2020-10-16T07:57:33.7722711Z"
    }
}

It's wield if we directly set the second property, it could work. But if we set the first property, then set the second property, it fails.

**Comment by blueww** [comment]
@zfchen95 
Would you please help to look at the 413 error?

**Comment by yupwei68** [comment]
On fact, after first update, the blob service could not be updated in 12 minutes or longer.

1. PUT /subscriptions/.../resourceGroups/acctestAzureRMSA-201028133158646092/providers/Microsoft.Storage/storageAccounts/unlikely23exst2acctqgubk/blobServices/default?api-version=2019-06-01 HTTP/1.1
{"properties":{"cors":{"corsRules":[{"allowedOrigins":["http://www.example.com"],"allowedMethods":["GET","PUT","PATCH"],"maxAgeInSeconds":500,"exposedHeaders":["x-tempo-*"],"allowedHeaders":["x-tempo-*"]}]},"deleteRetentionPolicy":{"enabled":true,"days":300},"isVersioningEnabled":true,"changeFeed":{"enabled":true},"restorePolicy":{"days":5,"enabled":true}}}

HTTP/2.0 200 OK
Cache-Control: no-cache
Content-Type: application/json
**Date: Wed, 28 Oct 2020 05:34:47 GMT**
Expires: -1
Pragma: no-cache
Server: Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: f347bf01-2468-bddd-5664-42986973def6
X-Ms-Ratelimit-Remaining-Subscription-Writes: 1191
X-Ms-Request-Id: 533d479d-9056-49a2-9cbf-a18fbf3b9c2b
X-Ms-Routing-Request-Id: SOUTHEASTASIA:20201028T053447Z:90f2e197-37fe-4673-a772-c88ba469c699

{"id":"/subscriptions/.../resourceGroups/acctestAzureRMSA-201028133158646092/providers/Microsoft.Storage/storageAccounts/unlikely23exst2acctqgubk/blobServices/default","name":"default","type":"Microsoft.Storage/storageAccounts/blobServices","properties":{"changeFeed":{"enabled":true},"restorePolicy":{"enabled":true,"days":5},"cors":{"corsRules":[{"allowedOrigins":["http://www.example.com"],"allowedMethods":["GET","PUT","PATCH"],"maxAgeInSeconds":500,"exposedHeaders":["x-tempo-*"],"allowedHeaders":["x-tempo-*"]}]},"deleteRetentionPolicy":{"enabled":true,"days":300},"isVersioningEnabled":true}}

2. PUT /subscriptions/.../resourceGroups/acctestAzureRMSA-201028133158646092/providers/Microsoft.Storage/storageAccounts/unlikely23exst2acctqgubk/blobServices/default?api-version=2019-06-01 HTTP/1.1
{"properties":{"cors":{"corsRules":[{"allowedOrigins":["http://www.example.com"],"allowedMethods":["GET"],"maxAgeInSeconds":2000000000,"exposedHeaders":["x-tempo-*","x-method-*"],"allowedHeaders":["*"]},{"allowedOrigins":["http://www.test.com"],"allowedMethods":["PUT"],"maxAgeInSeconds":1000,"exposedHeaders":["x-tempo-*"],"allowedHeaders":["*"]}]},"deleteRetentionPolicy":{"enabled":true,"days":7},"isVersioningEnabled":true,"changeFeed":{"enabled":false},"restorePolicy":{"enabled":false}}}

HTTP/2.0 412 Precondition Failed
Content-Length: 210
Cache-Control: no-cache
Content-Type: application/json
**Date: Wed, 28 Oct 2020 05:46:00 GMT**
Expires: -1
Pragma: no-cache
Server: Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: f347bf01-2468-bddd-5664-42986973def6
X-Ms-Ratelimit-Remaining-Subscription-Writes: 1199
X-Ms-Request-Id: 515221fd-ee3a-4697-a3d9-01c6537ace89
X-Ms-Routing-Request-Id: SOUTHEASTASIA:20201028T054601Z:ed275dc6-a00e-4531-af94-b4a86479a74e

{"error":{"code":"ContainerOperationFailure","message":"The condition specified using HTTP conditional header(s) is not met.\nRequestId:9630709c-101e-002e-01ed-ac9034000000\nTime:2020-10-28T05:46:01.0843639Z"}}

**Comment by aristosvo** [comment]
Is this already being addressed? Using the portal I'm able to change all kinds of settings within seconds, what is the difference?

**Comment by timja** [comment]
@yupwei68 / @zfchen95 is there any work going on to look at this?

Thanks :)

**Comment by zfchen95** [comment]
hi, the error seems a bit old. It happened in 2020. Do you have a more recent repro?

**Comment by aristosvo** [comment]
Yes, [here](https://github.com/hashicorp/terraform-provider-azurerm/pull/13267)

**Comment by matt-byrne** [comment]
This is still blocking https://github.com/hashicorp/terraform-provider-azurerm/issues/9020 which is a pretty important feature. 

Any update on this?

**Comment by zfchen95** [comment]
@blueww could you please help involve the right contact in the Storage team? I am not with Microsoft anymore.

**Comment by blueww** [comment]
@HimanshuChhabra , @priyapansrp 
Would you please help to look at this SRP server issue?

**Comment by ByMitta** [comment]
I am also encountering the same error. Though I am not using Terraform but Bicep which translates it to an ARM Template which is deployed using a Azure DevOps pipeline task.

The initial deploy of the storage account was successful. Another deploy after that I enabled PITR which was also successful. But the deploys after these, with no changes in the bicep/arm template are given me the same error now.

API Version: storageAccounts@2021-09-01
Bicep version: v0.9.1
Azure DevOps task: AzureResourceGroupDeployment@2
